app/src/main/java/com/example/tasky/feature_agenda/presentation/agenda_screen/AgendaScreen.kt

```diff
@@ -0,0 +1,119 @@
+package com.example.tasky.feature_agenda.presentation.agenda_screen
+
+import android.widget.Toast
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavController
+import com.example.tasky.feature_agenda.presentation.agenda_screen.components.DatePickerDialog
+import com.example.tasky.feature_agenda.presentation.agenda_screen.components.DayChip
+import com.example.tasky.feature_agenda.presentation.agenda_screen.components.TopSection
+import com.example.tasky.presentation.theme.BackgroundBlack
+import com.example.tasky.presentation.theme.BackgroundWhite
+import com.example.tasky.util.Result
+import com.example.tasky.util.Screen
+import com.vanpra.composematerialdialogs.rememberMaterialDialogState
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+
+@Composable
+fun AgendaScreen(
+    navController: NavController,
+    viewModel: AgendaViewModel = hiltViewModel()
+) {
+    viewModel.dateDialogState = rememberMaterialDialogState()
+    val state by viewModel.state.collectAsStateWithLifecycle()
+
+    val context = LocalContext.current
+
+    LaunchedEffect(key1 = true) {
+        viewModel.logoutResult.collect { result ->
+            when(result) {
+                is Result.Error -> Toast.makeText(context, result.message, Toast.LENGTH_LONG).show()
+                is Result.Success -> navController.navigate(Screen.LoginScreen.route)
+            }
+        }
+    }
```

Comment:
Strictly speaking, you need something like this to collect one-time events 100% safely without risking to lose some:
```
@Composable
private fun <T> ObserveAsEvents(flow: Flow<T>, onEvent: (T) -> Unit) {
    val lifecycleOwner = LocalLifecycleOwner.current
    LaunchedEffect(flow, lifecycleOwner.lifecycle) {
        lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
            withContext(Dispatchers.Main.immediate) {
                flow.collect(onEvent)
            }
        }
    }
}
```
For more info, I talk about the problem in this video: https://youtu.be/njchj9d_Lf8?si=0to4KwRHzEIECID8